app/domain/authentication/authn_jwt/identity_providers/create_identity_provider.rb

```diff
@@ -0,0 +1,41 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Factory for jwt identity providers.
+      # If Identity variable is configured factory return the decoded_token_provider
+      # If the Identity variable is not configured and there is account field in url factory returns url provider
+      # If the above conditions are not met exception is raised
+      CreateIdentityProvider = CommandClass.new(
+        dependencies: {
+          identity_from_url_provider_class: Authentication::AuthnJwt::IdentityProviders::IdentityFromUrlProvider,
+          identity_from_decoded_token_class: Authentication::AuthnJwt::IdentityProviders::IdentityFromDecodedTokenProvider,
```

Comment:
dependencies that are command classes should have `.new` as we want to initialize them only once. We use command_classes by calling their `call` function each time. thus, they can be initialzied once.

please go over all the files in this PR and verify that command classes are initialized when they are injected as dependencies

Reply:
and they also don't need the suffix `_class`. we use that for object classes so when we later call their `.new` method it makes sense

Reply:
This speicifc two classes are not command classes.
But i will go over the code and change dependencies that are command classes that we don't inject anything there to be with new

Reply:
Went over code and done